src/app/modules/app/app.js

```diff
@@ -123,18 +139,106 @@
                 });
             };
 
-                // At some point, we may manually control topLanguages, and have a 'more' link to show them all.
-//            $scope.topLanguages = [
-//                {isoCode: 'en', name:'English'},
-//                {isoCode: 'tpi', name:'Tok Pisin'},
-//                {isoCode: 'th', name:'Thai'},
-//                {isoCode: 'id', name:'Bahasia Indonesia'},
-//                {isoCode: 'fr', name:'French'}
-//            ];
             languageService.getLanguages().then(function(languages) {
-                $scope.topLanguages = languages;
+                //This is the number of displayed languages on the browse sidebar before (n more...)
+                var numberOfTopLanguages = 4;
+
+                function compareLang(a, b) {
+                    return a.name < b.name ? -1 : 1;
+                }
+
+                //If all languages can be held in the top list, just fill the top list
+                if(languages.length <= numberOfTopLanguages) {
+                    $scope.topLanguages = languages.sort(compareLang);
+                    $scope.otherLanguages = [];
+                }
+                else {
+                    //Split the list of languages into top and other, and sort the lists alphabetically
+                    $scope.topLanguages = languages.slice(0, numberOfTopLanguages).sort(compareLang);
+                    $scope.otherLanguages = languages.slice(numberOfTopLanguages, languages.length).sort(compareLang);
+                }
+
+                for(var i = 0; i < $scope.otherLanguages.length; i++) {
```

Comment:
Deserves a comment..."If the current language is in the "others" list move it to the "top/visible" list"?